### PR TITLE
Improve the CSS Animation API for $animateCss and JS Animations 

### PR DESCRIPTION
**Pull request by matsko**
I wrote this PR at the airport

### CODE REVIEW

File: test/ngAnimate/animateCssSpec.js [hunk 1]
```diff
@@ -316,6 +316,96 @@ describe("ngAnimate $animateCss", function() {
...
+          var animator = $animateCss(element, {
+            event: 'enter',
+            structural: true
+          });
+
+          expect(element).not.toHaveClass('ng-enter');
+          expect(animator.start).toBeTruthy();
+        }));
+
+        they("should close the animation, but still still accept $prop callbacks if no animation is detected",
```
Comment (LeonardoBraga) [inline test/ngAnimate/animateCssSpec.js]: "still still"

File: test/ngAnimate/animateCssSpec.js [hunk 2]
```diff
@@ -316,6 +316,96 @@ describe("ngAnimate $animateCss", function() {
...
+        it("should always return an object even if no animation is detected",
+          inject(function($animateCss) {
+
+          ss.addRule('.some-animation', 'background:red;');
+
+          element.addClass('some-animation');
+          var animator = $animateCss(element, options);
+
+          expect(animator).toBeTruthy();
+          expect(animator.start).toBeTruthy();
```
Comment (LeonardoBraga) [inline test/ngAnimate/animateCssSpec.js]: I'd suggest to use the test to enforce the API contract as well:
```js
expect(animator.start instanceof Function).toBe(true);
// or
expect(angular.isFunction(animator.start)).toBe(true);
```

The same would apply for the other methods being testes as `toBeTruthy`

**Comment by matsko** [comment]
@LeonardoBraga all fixed!

**Comment by IgorMinar** [comment]
the build failed. looks like a flake. I restarted it.

left a bunch of comments but overall this looks good. fix it up, get the build to pass and merge

File: test/ngAnimate/animateCssSpec.js (hunk 1 above)
Comment (matsko) [inline test/ngAnimate/animateCssSpec.js]: fixed.

**Comment by matsko** [comment]
Landed as https://github.com/angular/angular.js/commit/0681a5400e4150a961f9c8651e55623ca23b0cc2

**Pull request closed without merging**